### PR TITLE
Mesh Index-preserving Skeletonization and Masked Meshes

### PR DESCRIPTION
**Pull request by ceesem**
In this PR I introduce a couple of useful features that keep track of a lot of the index hell problems.

1) Skeletonization can return a mapping from mesh indices to skeleton indices
2) SkeletonForests, which build an array of single-component skeletons, can remap index lists from the vertex list they were given into their internal vertex organization. This could be made less clunky.
3) I introduce a MaskedMesh class that subclasses Mesh and makes it quick and easy to filter a mesh one or multiple times with any index- or boolean-based filter (e.g. only nodes in a graph component), do standard mesh computations, then remap resulting node indices back to the original mesh indexing. 
4) Replace most uses of scipy's cKDTree with the pykdtree package, which is much faster.
5) Various refactoring and some movement of utility functions.

Gotta get better at gradual PRs, I let my branch drift too much.

### CODE REVIEW

**Comment by ceesem** [comment]
Oh, and:
6) SWC export for skeletons. This needs to get adapted slightly to work for skeleton forests, but this shouldn't be too much work.

**Comment by fcollman** [comment]
@ceesem can you merge your changes with master so this can get merged?

**Comment by ceesem** [comment]
> @ceesem can you merge your changes with master so this can get merged?

Done.

File: meshparty/trimesh_io.py [hunk 1]
```diff
@@ -268,8 +270,8 @@ def _filename(self, seg_id):
         return "%s/%d.h5" % (self.disk_cache_path, seg_id)
 
     def mesh(self, filename=None, seg_id=None, cache_mesh=True,
-             merge_large_components=True, remove_duplicate_vertices=True,
-             overwrite_merge_large_components=False):
+             merge_large_components=False, remove_duplicate_vertices=False,
```
Comment (sdorkenw) [inline meshparty/trimesh_io.py]: Please keep the old defaults here

Comment (ceesem) [reply meshparty/trimesh_io.py]: Fixed

**Pull request merged**